pkg/apis/core/validation/validation.go

```diff
@@ -902,6 +902,21 @@ func validateGlusterfsVolumeSource(glusterfs *core.GlusterfsVolumeSource, fldPat
 	}
 	return allErrs
 }
+func validateGlusterfsPersistentVolumeSource(glusterfs *core.GlusterfsPersistentVolumeSource, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(glusterfs.EndpointsName) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("endpoints"), ""))
+	}
+	if len(glusterfs.Path) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("path"), ""))
+	}
+	if glusterfs.EndpointsNamespace != nil {
+		endpointNs := glusterfs.EndpointsNamespace
+		allErrs = append(allErrs, field.Required(fldPath.Child("endpointsNamespace"), ""))
```

Comment:
this should only be an error `if glusterfs.EndpointsNamespace == ""`... as is, this makes providing a namespace an unconditional error

Reply:
that also means we need a test for this :)

Reply:
Yeah :),  changed as suggested and added test.